### PR TITLE
Rewrite charm store docs to new charm tool

### LINKED ISSUE #937: Charm Store Workflow: Needs docs on how to push/publish charms/bundles to the store
This bug is to add instructions on how charm and bundle authors push/publish their charms and bundles to the juju charm store.

### PR DESCRIPTION
**Pull request by marcoceppi**
in process of landing in xenial, in meantime install charm from ppa:juju/devel and ppa:juju/stable

fixes #937

### CODE REVIEW

**Comment by chuckbutler** [comment]
+1 LGTM

Nice diagrams as well, and thanks for explaining channels in here. Solid work.

File: src/en/authors-charm-store.md [hunk 1]
```diff
@@ -2,204 +2,219 @@ Title: The Juju charm store
...
 
-This process is designed to allow prospective developers to have their charms
-reviewed and updated in the [Charm Store](https://jujucharms.com) in a timely
-manner that ensures peer reviews and quality.
+  - [Charm development information](developer-getting-started.html)
+  - [Bundle creation information](charms-bundles.html)
 
-## Submitting a new Charm
+In order to interact with the charm store you will need the latest
+[charm command](tools-charm-tools.html) and a charm store login.
```
Comment (urosj) [inline src/en/authors-charm-store.md]: For now, to login into charm store, you'll need to have both Ubuntu SSO and Launchpad (LP) account. Due to historical reasons, charm store users the username from LP.

File: src/en/authors-charm-store.md [hunk 2]
```diff
@@ -2,204 +2,219 @@ Title: The Juju charm store
...
-or to submit your charm for 14.04: 
-     
-```bash
-bzr push lp:~<your-lp-username>/charms/trusty/<your-charm>/trunk
-``` 
+## Entities explained
 
+When an entity, a charm or bundle, is pushed for the first time to the store it's
+placed as version 0 in the unpublished channel. Every revision of an entity lives
+in the unpublished channel. Subsequent pushes to the store will automatically
```
Comment (urosj) [inline src/en/authors-charm-store.md]: s/Subsequent pushes/Subsequent pushes of different content

File: src/en/authors-charm-store.md [hunk 3]
```diff
@@ -2,204 +2,219 @@ Title: The Juju charm store
...
+increment this number. So if an entity is pushed 4 times the revision history
```
Comment (urosj) [inline src/en/authors-charm-store.md]: s/an entity/a different entity

File: src/en/authors-charm-store.md [hunk 4]
```diff
@@ -2,204 +2,219 @@ Title: The Juju charm store
...
+would look as follows:
 
-Your charm should then be looked at in a timely manner.
+```
+entity: 0---1---2---3
+
+```
+
+Channels group revisions into named streams. There are currently to channels:
```
Comment (urosj) [inline src/en/authors-charm-store.md]: s/to channels/two channels

File: src/en/authors-charm-store.md [hunk 5]
```diff
@@ -2,204 +2,219 @@ Title: The Juju charm store
...
+stable and development. Each channel also tracks history of revisions. When a
+revision is published to a channel, that channel pointer is incremented and the
```
Comment (urosj) [inline src/en/authors-charm-store.md]: s/is incremented/is changed

**Comment by urosj** [comment]
Some comments, but otherwise LGTM, TY!

**Comment by marcoceppi** [comment]
Thanks for the feedback @urosj moving from WIP -> ready review

**Comment by evilnick** [comment]
:rice_cracker: 

@matthewhelmke perhaps you could give this page a once-over after merge :)

**Pull request merged**